src/test/HostActivationTests/FrameworkResolution/RollForwardSettings.cs

```diff
@@ -0,0 +1,277 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Cli.Build;
+using Microsoft.DotNet.Cli.Build.Framework;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.FrameworkResolution
+{
+    public class RollForwardSettings :
+        FrameworkResolutionBase,
+        IClassFixture<RollForwardSettings.SharedTestState>
+    {
+        private const string MiddleWare = "MiddleWare";
+
+        private SharedTestState SharedState { get; }
+
+        public RollForwardSettings(SharedTestState sharedState)
+        {
+            SharedState = sharedState;
+        }
+
+        // Verifies that default behavior is Minor
+        [Fact]
+        public void Default()
+        {
+            RunTest(
+                new TestSettings()
+                    .WithRuntimeConfigCustomizer(runtimeConfig => runtimeConfig
+                        .WithFramework(MicrosoftNETCoreApp, "4.0.0")))
+                .Should().Fail()
+                .And.DidNotFindCompatibleFrameworkVersion();
+
+            RunTest(
+                new TestSettings()
+                    .WithRuntimeConfigCustomizer(runtimeConfig => runtimeConfig
+                        .WithFramework(MicrosoftNETCoreApp, "5.0.0")))
+                .Should().Pass()
+                .And.HaveResolvedFramework(MicrosoftNETCoreApp, "5.1.3");
+        }
+
+        // Verifies that invalid values is checked in all settings locations
+        [Theory]
+        [InlineData(SettingLocation.CommandLine)]
+        [InlineData(SettingLocation.Environment)]
+        [InlineData(SettingLocation.RuntimeOptions)]
+        [InlineData(SettingLocation.FrameworkReference)]
+        public void InvalidValue(SettingLocation settingLocation)
+        {
+            RunTest(
+                new TestSettings()
+                    .WithRuntimeConfigCustomizer(runtimeConfig => runtimeConfig
+                        .WithFramework(MicrosoftNETCoreApp, "4.0.0"))
+                    .With(RollForwardSetting(settingLocation, "InvalidValue")))
+                .Should().Fail()
+                .And.DidNotRecognizeRollForwardValue("InvalidValue");
+        }
+
+        // Verifies that the value ignores casing on command line
```

Comment:
Why do we want to ignore case?  Seems simpler to no ignore case if it could be allowed.

Reply:
The problem is that the setting is accepted in several different places - command line, env. variable, runtime config. The natural casing for each is different. We define the enum as Pascal cased, but people are likely to use different casing on command line (all lowercase it likely), while in env. variable Pascal case or all upper case might feel more natural.
There's really no reason to be restrictive about casing here, we don't care, so it's better to leave it free - in my opinion. The original spec didn't define this part... and it's subjective … so I'm willing to change my mind if people think it would be better to be strict.

Reply:
The fact that there is a tests seems to indicate complexity.  I guess if there is an early `ToUpper()`/`ToLower()` it doesn't add much complexity. 

I am used to a case sensitive world. `C`/`C++`/`C#`/`python`/`file paths`/`git options`...

No strong opinion.